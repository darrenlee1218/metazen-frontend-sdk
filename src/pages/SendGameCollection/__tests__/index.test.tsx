import React from 'react';
import { render } from '@testing-library/react';
import { SendGameCollection } from '..';
import { ThemeProvider } from '@mui/material/styles';
import defaultTheme from '@theme/theme';

jest.mock('@redux/api/assetMetadataCache', () => ({
  __esModule: true,
  useGetAssetMetadataQuery: () => ({
    data: {
      data: {
        name: 'Nft Name',
        image: '404.png',
      },
    },
    isLoading: false,
  }),
}));

jest.mock('@hooks/useTokenByKey', () => ({
  __esModule: true,
  useTokenByKey: () => ({
    contractAddress: '0x0000000000000000000000000000000000000000',
  }),
}));

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useLocation: () => ({
    search: '',
  }),
  useParams: () => ({
    key: '',
  }),
  useNavigate: () => () => {},
}));

const expectedItems = 6;
const mockedBalancesData = Array.from({ length: expectedItems }).map((_, index) => ({
  nftTokenId: index,
}));
jest.mock('@redux/api/bookKeeping', () => ({
  __esModule: true,
  useGetBalancesByAssetIdsQuery: () => ({
    data: mockedBalancesData,
    isLoading: false,
  }),
}));

describe('Send Game Collection', () => {
  it('should render correctly', () => {
    const baseDom = render(
      <ThemeProvider theme={defaultTheme()}>
        <SendGameCollection />
      </ThemeProvider>,
    );
    expect(baseDom.baseElement).toMatchSnapshot();
  });

  it('should render the correct number of nft items', () => {
    const baseDom = render(
      <ThemeProvider theme={defaultTheme()}>
        <SendGameCollection />
      </ThemeProvider>,
    );
    expect(baseDom.getAllByText('Nft Name')).toHaveLength(expectedItems);
  });
});
