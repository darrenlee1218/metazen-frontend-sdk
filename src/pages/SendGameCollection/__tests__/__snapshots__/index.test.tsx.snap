// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send Game Collection should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1e4br2p"
    >
      <div
        class="MuiBox-root css-tfe0o3"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1rgr0y-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xtuuzo-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-16vochq-MuiTypography-root"
        >
          Send NFT
        </h2>
        <div
          class="MuiBox-root css-8h2w1h"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-s19m2i-MuiTypography-root"
      >
        Select which item you would like to send.
        <br />
        Only 1 item can be choosen.
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-8efvy8-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root MuiBox-root css-1lqyjo8-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-hjbv5f"
            >
              <img
                alt="Nft Name image"
                class="MuiBox-root css-djl8wu"
                src="404.png"
              />
            </div>
            <div
              class="MuiBox-root css-120alf"
              data-testid="asset-descriptor"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
              >
                Nft Name
              </h5>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root MuiBox-root css-1lqyjo8-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-hjbv5f"
            >
              <img
                alt="Nft Name image"
                class="MuiBox-root css-djl8wu"
                src="404.png"
              />
            </div>
            <div
              class="MuiBox-root css-120alf"
              data-testid="asset-descriptor"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
              >
                Nft Name
              </h5>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root MuiBox-root css-1lqyjo8-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-hjbv5f"
            >
              <img
                alt="Nft Name image"
                class="MuiBox-root css-djl8wu"
                src="404.png"
              />
            </div>
            <div
              class="MuiBox-root css-120alf"
              data-testid="asset-descriptor"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
              >
                Nft Name
              </h5>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root MuiBox-root css-1lqyjo8-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-hjbv5f"
            >
              <img
                alt="Nft Name image"
                class="MuiBox-root css-djl8wu"
                src="404.png"
              />
            </div>
            <div
              class="MuiBox-root css-120alf"
              data-testid="asset-descriptor"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
              >
                Nft Name
              </h5>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root MuiBox-root css-1lqyjo8-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-hjbv5f"
            >
              <img
                alt="Nft Name image"
                class="MuiBox-root css-djl8wu"
                src="404.png"
              />
            </div>
            <div
              class="MuiBox-root css-120alf"
              data-testid="asset-descriptor"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
              >
                Nft Name
              </h5>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root MuiBox-root css-1lqyjo8-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiBox-root css-hjbv5f"
            >
              <img
                alt="Nft Name image"
                class="MuiBox-root css-djl8wu"
                src="404.png"
              />
            </div>
            <div
              class="MuiBox-root css-120alf"
              data-testid="asset-descriptor"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
              >
                Nft Name
              </h5>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1mphh29-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-ea6zr6"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth Mui-disabled css-20f9f1-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          disabled=""
          id="mui-1"
          tabindex="-1"
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</body>
`;
